pasqal_cloud/device/configuration/base_config.py

```diff
@@ -26,7 +26,7 @@ class BaseConfig:
     parameters are added to the configuration dictionary.
     """
 
-    extra_config: Optional[dict[str, Any]] = None
```

Comment:
any reason for this? I'm not sure what convention we follow, but the most recent standard says to favour `dict` as the type over `Dict`.

Reply:
Not using the `Dict` type hint in a pydantic model for python version <3.8 raises errors. See for example CI https://github.com/pasqal-io/pasqal-cloud/pull/81/checks?check_run_id=14051565461

Rule of thumb I am using is to use new type hints (e.g. `dict` and `list`) everywhere except for attributes of a pydantic model

Reply:
I see. Pulser has recently dropped support for 3.7, and it reaches of life in less than 3 weeks (27 Jun 2023).

I'll open an issue for that